pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go

```diff
@@ -24,7 +30,209 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
+		return nil
+	}
+
+	for _, backend := range conf.Backends {
+		configurer := plugin_xds.Configurer{
+			Mesh:               dataplane.GetMeta().GetMesh(),
+			TrafficDirection:   envoy.TrafficDirectionInbound,
+			SourceService:      mesh_proto.ServiceUnknown,
+			DestinationService: serviceName,
+			Backend:            backend,
+			Dataplane:          dataplane,
+		}
+
+		for _, chain := range listener.FilterChains {
+			if err := configurer.Configure(chain); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func configureOutbound(
+	toRules xds.ToRules,
+	dataplane *core_mesh.DataplaneResource,
+	subset xds.Subset,
+	destinationServiceName string,
+	listener *envoy_listener.Listener,
+) error {
+	sourceService := dataplane.Spec.GetIdentifyingService()
+
+	var conf *api.MeshAccessLog_Conf
+	if computed := toRules.Rules.Compute(subset); computed != nil {
+		conf = computed.(*api.MeshAccessLog_Conf)
+	} else {
+		return nil
+	}
+
+	for _, backend := range conf.Backends {
+		configurer := plugin_xds.Configurer{
+			Mesh:               dataplane.GetMeta().GetMesh(),
+			TrafficDirection:   envoy.TrafficDirectionOutbound,
+			SourceService:      sourceService,
+			DestinationService: destinationServiceName,
+			Backend:            backend,
+			Dataplane:          dataplane,
+		}
+
+		for _, chain := range listener.FilterChains {
+			if err := configurer.Configure(chain); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func applyToInbounds(
+	rules xds.FromRules, inboundListeners map[xds.InboundListener]*envoy_listener.Listener, dataplane *core_mesh.DataplaneResource,
+) error {
+	for _, inbound := range dataplane.Spec.GetNetworking().GetInbound() {
+		iface := dataplane.Spec.Networking.ToInboundInterface(inbound)
+
+		listenerKey := xds.InboundListener{
+			Address: iface.DataplaneIP,
+			Port:    iface.DataplanePort,
+		}
+		listener, ok := inboundListeners[listenerKey]
+		if !ok {
+			continue
+		}
+
+		serviceName := inbound.GetTags()[mesh_proto.ServiceTag]
+
+		return configureInbound(rules.Rules[listenerKey], dataplane, xds.MeshService(serviceName), listener)
+	}
+	return nil
+}
+
+func applyToOutbounds(
+	rules xds.ToRules, outboundListeners map[mesh_proto.OutboundInterface]*envoy_listener.Listener, dataplane *core_mesh.DataplaneResource,
+) error {
+	for _, outbound := range dataplane.Spec.Networking.GetOutbound() {
+		oface := dataplane.Spec.Networking.ToOutboundInterface(outbound)
+
+		listener, ok := outboundListeners[oface]
+		if !ok {
+			continue
+		}
+
+		serviceName := outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+
+		if err := configureOutbound(rules, dataplane, xds.MeshService(serviceName), serviceName, listener); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func applyToTransparentProxyListeners(
+	policies xds.TypedMatchingPolicies, ipv4 *envoy_listener.Listener, ipv6 *envoy_listener.Listener, dataplane *core_mesh.DataplaneResource,
+) error {
+	// TODO inbound listener?
```

Comment:
🤔

Is there anything to do here?

Reply:
I'm not completely sure yet. We didn't log inbound traffic before but now we can set `from` rules so _something_ should probably happen with the inbound passthrough listeners.

Reply:
it's a separate issue then?

Reply:
separate PR at least